### DISCUSSION TITLE
Cookie ⚡️ Node.js с примерами кода

### DISCUSSION THREAD

**Discussion by giscus** [xsltdev/nodejsdev.ru#53] [Announcements]
# Cookie ⚡️ Node.js с примерами кода

Справочник Node.js с примерами кода

https://nodejsdev.ru/guides/webdraftt/cookie/?ysclid=lp82pu4if7132690777

**Comment by vokmar** [comment]
maxAge - определяет время жизни файла в секундах;
точно секунды?
проверьте.

**Reply by bndby** [reply]
Вы правы, в миллисекундах.
Пофиксил.

**Comment by GenGPT** [comment]
//index.html
<!DOCTYPE html>

  <html lang="en">

  <head>

    <meta charset="UTF-8">

    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <title>Бронирование отелей</title>

    <link rel="stylesheet" href="style.css">

  </head>

  <body>

    <h1>Добро пожаловать! Зарегистрируйтесь или войдите:</h1>

    <nav>

      <a href="register.html">Зарегистрироваться</a>

      <a href="login.html">Войти</a>

    </nav>

  </body>

  </html>


//login.html

<!DOCTYPE html>

<html lang="en">

<head>

  <meta charset="UTF-8">

  <meta name="viewport" content="width=device-width, initial-scale=1.0">

  <title>Вход</title>

  <link rel="stylesheet" href="style.css">

</head>

<body>

  <h1>ВХОД</h1>

  <form id="loginForm">

    <label for="username">Логин:</label>

    <input type="text" id="username" name="username" required>

    <label for="password">Пароль:</label>

    <input type="password" id="password" name="password" required>

    <button type="submit">ВОЙТИ</button>

  </form>

  <script src="main.js"></script>

</body>

</html>


//registration.html

<!DOCTYPE html>

<html lang="en">

<head>

  <meta charset="UTF-8">

  <meta name="viewport" content="width=device-width, initial-scale=1.0">

  <title>Регистрация</title>

  <link rel="stylesheet" href="style.css">

</head>

<body>

  <h1>РЕГИСТРАЦИЯ</h1>

  <form id="registerForm">

    <label for="username">Логин:</label>

    <input type="text" id="username" name="username" required>

    <label for="password">Пароль:</label>

    <input type="password" id="password" name="password" required>

    <button type="submit">Зарегистрироваться</button>

  </form>

  <script src="main.js"></script>

</body>

</html>


//bookings.html
<!DOCTYPE html>

<html lang="en">

<head>

  <meta charset="UTF-8">

  <meta name="viewport" content="width=device-width, initial-scale=1.0">

  <title>Bookings</title>

  <link rel="stylesheet" href="style.css">

</head>

<body>

  <h1>Bookings</h1>

  <form id="bookingForm">

    <label for="room_id">Room ID:</label>

    <input type="number" id="room_id" name="room_id" required>

    <label for="guest_name">Guest Name:</label>

    <input type="text" id="guest_name" name="guest_name" required>

    <label for="check_in_date">Check-in Date:</label>

    <input type="date" id="check_in_date" name="check_in_date" required>

    <label for="check_out_date">Check-out Date:</label>

    <input type="date" id="check_out_date" name="check_out_date" required>

    <button type="submit">Book</button>

  </form>

  <h2>Your Bookings</h2>

  <ul id="bookingsList"></ul>

  <script src="main.js"></script>

</body>

</html>

**Comment by YuraKissil** [comment]
Было бы неплохо упомянуть, что "cookieParser = require('cookie-parser');" и "app.use(cookieParser('secret key'));" требуются только для signed cookie. Иначе при попытке установить получим "Error: cookieParser("secret") required for signed cookies".
Кстати, у меня в предложенной реализации при запросе /get-cookie не выводились signed cookies, установленные ранее в запросе /set-cookie

**Comment by YuraKissil** [comment]
Да и фраза "Первым параметром функции промежуточной обработки cookie-parser передается секретный ключ, который используется в целях безопасности для шифрования значений файлов перед их отправкой клиенту.", как-то не подтвердилась. По факту в куке читаемо и само значение, а так же есть и подпись. То есть "шифрования значений" здесь трактуется очень своеобразно,

**Comment by YuraKissil** [comment]
Для чтения при запросе /get-cookie ранее установленных signed cookies успешно использовал req.signedCookies, найденный https :// expressjs.com /en/api.html#res.cookie , а не просто req.сookies

**Comment by YuraKissil** [comment]
В примере "maxAge: 3600 * 24," лучше дополнить до "maxAge: 1000 * 3600 * 24" - так больше похоже на милисекунды
